src/server/ua_services_subscription.c

```diff
@@ -403,22 +404,23 @@ Operation_SetMonitoringMode(UA_Server *server, UA_Session *session,
     if(mon->monitoringMode == smc->monitoringMode)
         return;
 
-    mon->monitoringMode = smc->monitoringMode;
     if(mon->monitoringMode == UA_MONITORINGMODE_REPORTING) {
```

Comment:
Shouldn't this be 'if(smc->monitoringMode == ...)`? Same for the next if?
smc contains the new monitoring mode, so if the mode is changed, e.g. set to disabled, the queue should be deleted, and not if the monitored item is already deleted? Therefore the previous code was correct (except the val->data stuff), or am I wrong?